helm/configs/config.yaml

```diff
@@ -0,0 +1,281 @@
+
+  # OIDC issuer advertised by the service. Defaults to `public_base`
+  issuer: {{ .Values.config.http.issuer | default .Values.config.http.public_base | quote }}
+
+  # Each listener can serve multiple resources, and listen on multiple TCP ports or UNIX sockets.
+  # List of HTTP listeners, see below
+  listeners:
+    # The name of the listener, used in logs and metrics
+    - name: web
+
+      # List of resources to serve
+      resources: {{- range .Values.config.listeners.resources }}
+      - name: {{ .name }}
+      {{- if .path }}
+        path: {{ .path -}}
+      {{- end }}
+      {{- if .playground }}
+        playground: {{ .playground }}
+      {{- end }}
+      {{- end }}
+      - name: health
+        path: /health
+      {{- if .Values.telemetry.metrics.enabled }}
+      - name: prometheus
+        path: /health
+      {{- end }}
+
+      binds:
+        - address: '[::]:8080'
+  proxy_protocol: {{ .Values.config.listeners.proxy_protocol }}
+  trusted_proxies: {{ range .Values.config.trusted_proxies }}
+    - {{ . }}
+    {{- end }}
+
+  # Configure how to connect to the PostgreSQL database.
+database:
+  # Full connection string as per
+  # https://www.postgresql.org/docs/13/libpq-connect.html#id-1.7.3.8.3.6
+  #uri: postgresql://user:password@hostname:5432/database?sslmode=require
+
+  # -- OR --
+  # Separate parameters
+  host: {{ .Values.database.host }}
+  port: {{ .Values.database.port }}
+  #socket:
+  username: {{ .Values.database.username }}
+  password: {{ .Values.database.password }}
+  database: {{ .Values.database.dbName }}
+
+  # Additional parameters for the connection pool
+  min_connections: {{ .Values.database.min_connections }}
+  max_connections: {{ .Values.database.max_connections }}
+  connect_timeout: {{ .Values.database.connect_timeout }}
+  idle_timeout: {{ .Values.database.idle_timeout }}
+  max_lifetime: {{ .Values.database.max_lifetime }}
+
+  # Settings related to the connection to the Matrix homeserver
+matrix:
+  # The homeserver name, as per the `server_name` in the Synapse configuration file
+  homeserver: {{ .Values.matrix.homeserver }}
+
+  # Shared secret used to authenticate the service to the homeserver
+  # This must be of high entropy, because leaking this secret would allow anyone to perform admin actions on the homeserver
+  secret: {{ .Values.matrix.sharedSecret | quote }}
+
+  # URL to which the homeserver is accessible from the service
+  endpoint: {{ .Values.matrix.homeserverUrl | quote }}
+
+  # Allows loading custom templates
+{{- if .Values.templates.enabled }}
+templates:
+  path: {{ .Values.templates.path }}
+  assets_manifest: {{ .Values.templates.assets }}
+{{- end }}
+
+  # List of OAuth 2.0/OIDC clients and their keys/secrets. Each client_id must be a ULID.
+clients:
+  # Confidential client
+  - client_id: {{ .Values.clients.client_id }}
+    client_auth_method: {{ .Values.clients.client_auth_method }}
+    client_secret: {{ .Values.clients.client_secret }}
```

Comment:
i understand that secrets handling mostly depends on custom integrations, but having secrets in a configmap is not so typical. [anance's synapse chart](https://gitlab.com/ananace/charts/-/blob/master/charts/matrix-synapse/templates/deployment.yaml?ref_type=heads#L69) for instance puts them into separate config files from secrets. it has pros and cons.